src/MediaStreamTrack.ts

```diff
@@ -107,7 +119,16 @@ export default class MediaStreamTrack extends EventTarget<MediaStreamTrackEventM
             throw new Error('Only implemented for video tracks');
         }
 
-        WebRTCModule.mediaStreamTrackSwitchCamera(this.id);
+        const constraints = deepClone(this._settings);
+
+        delete constraints.deviceId;
+        constraints.facingMode = this._settings.facingMode === 'user' ? 'environment' : 'user';
+
+        const switchImpl = async () => {
+            await this.applyConstraints(constraints);
+        };
+
+        switchImpl();
```

Comment:
Since we are not really awaiting this, we can simplee call `this.applyConstraints(constraints);` here.